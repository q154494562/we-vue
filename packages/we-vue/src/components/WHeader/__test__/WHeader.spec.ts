import { shallowMount } from '@vue/test-utils'
import Header from '../'

describe('header', () => {
  test('create', () => {
    const wrapper = shallowMount(Header, {
      propsData: {},
    })

    expect(wrapper.name()).toBe('w-header')
    expect(wrapper.classes()).toContain('wv-header')

    expect(wrapper.html()).toMatchSnapshot()

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('click header', () => {
    const wrapper = shallowMount(Header, {
      propsData: {},
    })

    wrapper.trigger('click')

    expect(wrapper.emitted('headerClick')).toBeTruthy()
  })
})
