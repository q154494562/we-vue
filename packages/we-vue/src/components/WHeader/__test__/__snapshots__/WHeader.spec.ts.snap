// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header create 1`] = `
<header class="wv-header is-fixed" style="background-color: rgb(33, 41, 44);">
  <div class="wv-header-btn left"></div>
  <div class="wv-header-title"></div>
  <div class="wv-header-btn right"></div>
</header>
`;

exports[`header create 2`] = `
<header class="wv-header is-fixed" style="background-color: rgb(33, 41, 44);">
  <div class="wv-header-btn left"></div>
  <div class="wv-header-title"></div>
  <div class="wv-header-btn right"></div>
</header>
`;
